Tools/TestWebKitAPI/Tests/WebKitCocoa/WebPushDaemon.mm

```diff
@@ -2381,29 +2384,35 @@ static size_t expectedSuccessfulMessages()
     });
     TestWebKitAPI::Util::run(&done);
 
+    // Verify that even after having sent a push message to the daemon, there are no pending messages, as it was already handled.
     done = false;
     [dataStore _getPendingPushMessages:^(NSArray<NSDictionary *> *messages) {
-        EXPECT_EQ(messages.count, 1u);
-
-        [dataStore _processPushMessage:messages.firstObject completionHandler:^(bool handled) {
-            EXPECT_TRUE(handled);
-            EXPECT_TRUE([delegate.get().mostRecentNotification.get().userInfo[@"WebNotificationDefaultActionURLKey"] isEqualToString:@"https://example.com/"]);
-            EXPECT_EQ(delegate.get().mostRecentAppBadge, 18446744073709551615ULL);
-            done = true;
-        }];
+        EXPECT_EQ(messages.count, 0u);
+        done = true;
     }];
     TestWebKitAPI::Util::run(&done);
 
+#if HAVE(FULL_FEATURED_USER_NOTIFICATIONS)
+    auto configuration = adoptNS([[_WKWebPushDaemonConnectionConfiguration alloc] init]);
```

Comment:
Would be nice to not hide the smart pointers behind `auto` in new code and use `RetainPtr configuration ` here.